src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+    let newSeparator = separator;
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        newSeparator = '';
+      }
```

Comment:
Why do you need this condition? You need just to add a separator only after elements under indexes less than `length - 1`.